addons/web/static/src/js/fields/signature.js

```diff
@@ -0,0 +1,296 @@
+odoo.define('web.Signature', function (require) {
+    "use strict";
+
+    var core = require('web.core');
+    var session = require('web.session');
+    var utils = require('web.utils');
+    var field_utils = require('web.field_utils');
+    var Dialog = require('web.Dialog');
+    var AbstractFieldBinary = require('web.basic_fields').AbstractFieldBinary;
+    var registry = require('web.field_registry');
+    var NameAndSignature = require('web.name_and_signature').NameAndSignature;
+
+
+    var qweb = core.qweb;
+    var _t = core._t;
+
+// The goal of this dialog is to ask the user a signature request.
+// It uses @see SignNameAndSignature for the name and signature fields.
+var SignatureDialog = Dialog.extend({
+    template: 'web.signature_dialog',
+    xmlDependencies: Dialog.prototype.xmlDependencies.concat(
+        ['/web/static/src/xml/name_and_signature.xml']
+    ),
+    custom_events: {
+        'signature_changed': '_onChangeSignature',
+    },
+
+    //----------------------------------------------------------------------
+    // Public
+    //----------------------------------------------------------------------
+
+    /**
+     * Allows options.
+     *
+     * @constructor
+     * @param {Widget} parent
+     * @param {Object} options
+     * @param {string} [options.title='Adopt Your Signature'] - modal title
+     * @param {string} [options.size='medium'] - modal size
+     * @param {Object} [options.nameAndSignatureOptions={}] - options for
+     *  @see NameAndSignature.init()
+     */
+    init: function (parent, options) {
+        options = options || {};
+
+        options.title = options.title || _t("Adopt Your Signature");
+        options.size = options.size || 'medium';
+        options.technical = false;
+
+        if (!options.buttons) {
+            options.buttons = [];
+            options.buttons.push({text: _t("Adopt and Sign"), classes: "btn-primary", disabled: true, click: function (e) {
+                this.confirmFunction();
+            }});
+            options.buttons.push({text: _t("Cancel"), close: true});
+        }
+
+        this._super(parent, options);
+
+        this.confirmFunction = function () {};
+
+        this.nameAndSignature = new NameAndSignature(this, options.nameAndSignatureOptions);
+    },
+    /**
+     * Inserts the name and signature.
+     *
+     * @override
+     */
+    start: function () {
+        this.$primaryButton = this.$footer.find('.btn-primary');
+        this.nameAndSignature.replace(this.$('.o_web_sign_name_and_signature'));
+        return this._super.apply(this, arguments);
```

Comment:
`return this._super.apply(this, arguments).then(function() {
               return this.nameAnd....replace(...)
 }`